File: frontend/lib/models/schedule.dart
class Lesson {
  final String time;
  final String roomCode;
  final String title;

  Lesson({required this.time, required this.roomCode, required this.title});

  factory Lesson.fromJson(Map<String, dynamic> json) => Lesson(
        time: json['time'],
        roomCode: json['room_code'],
        title: json['title'],
      );
}

class Schedule {
  final int userId;
  final List<Lesson> lessons;

  Schedule({required this.userId, required this.lessons});

  factory Schedule.fromJson(Map<String, dynamic> json) => Schedule(
        userId: json['user_id'],
        lessons: (json['lessons'] as List).map((e) => Lesson.fromJson(e)).toList(),
      );
}

File: backend/app/api/endpoints/schedule.py
from fastapi import APIRouter, Depends
from ..dependencies import get_current_user
from ...schemas import Schedule, Lesson

router = APIRouter()

@router.get('/schedule/{user_id}', response_model=Schedule)
def get_schedule(user_id: int, current_user=Depends(get_current_user)):
    # TODO: integrate with MISIS schedule API
    lessons = [
        Lesson(time="10:00", room_code="A-101", title="Math"),
        Lesson(time="12:00", room_code="B-202", title="Physics"),
    ]
    return Schedule(user_id=user_id, lessons=lessons)
